extension/js/common/core/crypto/pgp/pgp-hash.ts

```diff
@@ -16,9 +17,11 @@ export class PgpHash {
   };
 
   private static cryptoHashSha256Loop = async (string: string, times = 100000) => {
+    console.log(`cryptoHashSha256Loop start`);
     for (let i = 0; i < times; i++) {
       string = await PgpHash.sha256UtfStr(string);
     }
+    console.log(`cryptoHashSha256Loop start ${string}`);
```

Comment:
Regarding #4688, why do we loop 100000 times and try to get 100000th PgpHashed string? Is it because of security?
It seems too heavy operation.

@tomholub Why to we loop

Reply:
As I checked deeply, #4688 issue happens because it takes too long time (maybe 30-60secs in semaphoreci) to get `cryptoHashSha256Loop` value as it tries to run `PgpHash.sha256UtfStr` function 100000 times.

Reply:
It's called iterated hashing, and makes it somewhat harder to brute-force passwords in case they are not so strong. But what you found is interesting. In normal use, it takes about a second or so. You're saying that when run in puppeteer, the result is MUCH slower. Which is a problem for testing.

Reply:
Ah - looks like we run into this following / due to https://github.com/FlowCrypt/flowcrypt-browser/issues/4626 , haven't we?

Reply:
It's not slow in puppeteer. if i run in local it runs in a few seconds.
But in semaphoreci, it took longer than 20 secs.

Reply:
Correct

Reply:
Can you reproduce this slowness when testing manually without puppeteer? (you will need a non-enterprise account to test on)

Reply:
maybe window.crypto is slower than opgp.crypto in semaphoreci

Reply:
Anyway, given that we'd like to do a release, we could consider reverting #4626, which should fix it, and then having a closer look after a release.

Reply:
It takes about 3-5 seconds to send password protected emails (which uses this method)
> Can you reproduce this slowness when testing manually without puppeteer? (you will need a non-enterprise account to test on)

Reply:
maybe we can increase delay (waiting time) specially for this failed test?

Reply:
Ok. So it's slow in semaphore ci, but not horribly slow locally, even when using puppeteer. And it only happens after our change. Sounds indeed our code is less efficient than what was there before.

You can try to move the test to `flaky` test group. These tests run in series instead of parallel, so the CPU will be available for it. (normal tests run 20 different tests in parallel)

Later we'll still have to investigate further.

Reply:
ok